### PR TITLE
feat: set_id method for the node

### PR DESCRIPTION
**Pull request by arifBurakDemiray**

### CODE REVIEW

**Comment by turtledreams** [comment]
@AliRKat you gotta lint this

**Comment by arifBurakDemiray** [comment]
@AliRKat You are best. I have also added DeviceIdType enums. Well, if you would like to add tests, I really appreciate it. 😅. However, let me try to add some, if I cannot make it, I will let you know.

**Review by AliRKat — APPROVED** [review]
Did some cosmetic updates because I am trying to make tests look in a similar structure throughout the SDK. Cool work my brother! Not seeing anything missing anymore

File: lib/countly.js [hunk 1]
```diff
@@ -627,6 +628,26 @@ Countly.Bulk = Bulk;
         return Countly.device_id;
     };
 
+    /**
+    * Changes the current device ID according to the device ID type (the preffered method)
+    * @param {string} newId - new user/device ID to use. Must be a non-empty string value. Invalid values (like null, empty string or undefined) will be rejected
+    * */
+    Countly.set_id = function(newId) {
+        cc.log(cc.logLevelEnums.INFO, `set_id, Changing the device ID to: [${newId}]`);
```
Comment (turtledreams) [inline lib/countly.js:636]: lets add type check here also

**Review by turtledreams — CHANGES_REQUESTED** [review]

File: lib/countly.js (hunk 1 above)
Comment (arifBurakDemiray) [reply lib/countly.js:636]: which type check? should we move it to the upper

**Review by turtledreams — APPROVED** [review]

**Comment by turtledreams** [comment]
ah needs more lintin @AliRKat

**Pull request merged**